### PR TITLE
feat/Zklend_models 

### PR DESCRIPTION
**Pull request by binayak9932**
Issue  #272 

- [X]   Created new models for missing events [here](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/db/models/zklend_events.py)Inherited from [EventBaseModel](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/db/models/event.py#L19C7-L19C21)
Created new models for next events:
- [X]  RepaymentEventData
- [X]  DepositEventData
- [X]  BorrowingEventData
- [X]  WithdrawalEventData
- [X]  CollateralEnabledDisabledEventData
- [X]  Generated migration file
- [X] Added changes in the older migration beacuse there was an error during migration "Alembic was attempting to drop an index, ix_notification_email, which didnt  exist in the database." so i  wrapped it in a conditional check to first check  if the notification table exists before attempting to drop the index.

### CODE REVIEW

**Comment by binayak9932** [comment]
@djeck1432  Please review

File: apps/data_handler/db/models/zklend_events.py [hunk 1]
```diff
@@ -39,3 +39,80 @@ class LiquidationEventData(EventBaseModel):
...
+
+class RepaymentEventData(EventBaseModel):
+    """
+    Database model for Repayment event, inheriting from EventBaseModel.
+
+    This model stores details of a repayment event, including the addresses involved
+    and the amounts in raw and face formats.
+    """
+
+    __tablename__ = "repayment_event_data"
```
Comment (djeck1432) [inline apps/data_handler/db/models/zklend_events.py:52]: we don't need to have here `_data`. please remove it
`repayment_event` will be okay

File: apps/data_handler/db/models/zklend_events.py [hunk 2]
```diff
@@ -39,3 +39,80 @@ class LiquidationEventData(EventBaseModel):
...
+
+class DepositEventData(EventBaseModel):
+    """
+    Database model for Deposit event, inheriting from EventBaseModel.
+
+    This model stores details of a deposit event, including the user address,
+    token, and the face amount of the deposit.
+    """
+
+    __tablename__ = "deposit_event_data"
```
Comment (djeck1432) [inline apps/data_handler/db/models/zklend_events.py:69]: we don't need to have here `_data`. please remove it
`deposit_event` will be okay

File: apps/data_handler/db/models/zklend_events.py [hunk 3]
```diff
@@ -39,3 +39,80 @@ class LiquidationEventData(EventBaseModel):
...
+
+class BorrowingEventData(EventBaseModel):
+    """
+    Database model for Borrowing event, inheriting from EventBaseModel.
+
+    This model stores details of a borrowing event, including the user address,
+    token, and amounts in raw and face formats.
+    """
+
+    __tablename__ = "borrowing_event_data"
```
Comment (djeck1432) [inline apps/data_handler/db/models/zklend_events.py:84]: we don't need to have here ` _data.` please remove it
`borrowing_event` will be okay

File: apps/data_handler/db/models/zklend_events.py [hunk 4]
```diff
@@ -39,3 +39,80 @@ class LiquidationEventData(EventBaseModel):
...
+
+class WithdrawalEventData(EventBaseModel):
+    """
+    Database model for Withdrawal event, inheriting from EventBaseModel.
+
+    This model stores details of a withdrawal event, including the user address,
+    token, and the amount withdrawn.
+    """
+
+    __tablename__ = "withdrawal_event_data"
```
Comment (djeck1432) [inline apps/data_handler/db/models/zklend_events.py:100]: we don't need to have here  _data. please remove it
`withdrawal_event` will be okay

File: apps/data_handler/db/models/zklend_events.py [hunk 5]
```diff
@@ -39,3 +39,80 @@ class LiquidationEventData(EventBaseModel):
...
+
+class CollateralEnabledDisabledEventData(EventBaseModel):
+    """
+    Database model for CollateralEnabled/Disabled event, inheriting from EventBaseModel.
+
+    This model stores details of a collateral enabled/disabled event, including
+    the user address and the token involved.
+    """
+
+    __tablename__ = "collateral_enabled_disabled_event_data"
```
Comment (djeck1432) [inline apps/data_handler/db/models/zklend_events.py:115]: we don't need to have here `_data`. please remove it
`collateral_enabled_disabled_event` will be okay

**Review by djeck1432 — CHANGES_REQUESTED** [review]
1. Rework migration file after your changes
2. Remove `poetry.lock` if you didn't do any changes in `pyproject.toml`

**Comment by binayak9932** [comment]
@djeck1432  Please review

**Comment by binayak9932** [comment]
> 1. Rework migration file after your changes
> 2. Remove `poetry.lock` if you didn't do any changes in `pyproject.toml`

Done

**Review by djeck1432 — APPROVED** [review]
Looks good to me, good job

**Pull request merged**